### PR TITLE
Scrub Latent Dirichlet Allocation Transform (Just Renaming)

### PR DESCRIPTION
**Pull request by wschin**
Toward #2832. This PR handles Latent Dirichlet Allocation.

### CODE REVIEW

File: test/Microsoft.ML.StaticPipelineTesting/StaticPipeTests.cs [hunk 1]
```diff
@@ -675,7 +675,7 @@ public void LdaTopicModel()
             var est = data.MakeNewEstimator()
                 .Append(r => (
                     r.label,
-                    topics: r.text.ToBagofWords().ToLdaTopicVector(numTopic: 3, numSummaryTermPerTopic:5, alphaSum: 10, onFit: m => ldaSummary = m.LdaTopicSummary)));
+                    topics: r.text.ToBagofWords().ToLatentDirichletAllocationTopicVector(numberOfTopics: 3, numberOfSummaryTermsPerTopic:5, alphaSum: 10, onFit: m => ldaSummary = m.LdaTopicSummary)));
```
Comment (Ivanidzo4ka) [inline test/Microsoft.ML.StaticPipelineTesting/StaticPipeTests.cs]: >ToLatentDirichletAllocationTopicVector [](start = 50, length = 38)

It's not what I care much about Static pipeline right now, but I would call it `LatentDirichletAllocation` same as we use in Dynamic pipeline.

Comment (wschin) [reply test/Microsoft.ML.StaticPipelineTesting/StaticPipeTests.cs]: Why not. :)

---
In reply to: [263972927](https://github.com/dotnet/machinelearning/pull/2890#discussion_r263972927) [](ancestors = 263972927)

File: src/Microsoft.ML.Transforms/Text/TextCatalog.cs [hunk 2]
```diff
@@ -600,20 +600,20 @@ public static class TextCatalog
         public static LatentDirichletAllocationEstimator LatentDirichletAllocation(this TransformsCatalog.TextTransforms catalog,
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/Text/TextCatalog.cs]: >LatentDirichletAllocation [](start = 57, length = 25)

Whole text catalog is slightly different from any other catalogs.
In any other catalog, we expose constructor with few major options, and then just columnOptions array.
In this specific case we expose EVERYTHING.
Shall we cut it to bare minimum?

Comment (wschin) [reply src/Microsoft.ML.Transforms/Text/TextCatalog.cs]: Ok.

---
In reply to: [263974378](https://github.com/dotnet/machinelearning/pull/2890#discussion_r263974378) [](ancestors = 263974378)

File: src/Microsoft.ML.Transforms/Text/TextCatalog.cs [hunk 3]
```diff
@@ -600,20 +593,22 @@ public static class TextCatalog
...
```
Comment (sfilipi) [inline src/Microsoft.ML.Transforms/Text/TextCatalog.cs]: >LatentDirichletAllocation [](start = 57, length = 25)

not directly related to your change, but the other names are usually verbs. If we wanted to keep to that, shall this be named ApplyLatentDirichletAllocation

Comment (wschin) [reply src/Microsoft.ML.Transforms/Text/TextCatalog.cs]: Not sure. `ConvertLatentDirichletAllocation` looks a bit confusing. For example, we don't say `ConvertToMatrixFactorization`. One proper (mathematically) choice is `ProduceConvertLatentDirichletAllocationVector` but it's just too long..

---
In reply to: [264332826](https://github.com/dotnet/machinelearning/pull/2890#discussion_r264332826) [](ancestors = 264332826)

**Pull request merged**